### PR TITLE
TOTP + value types

### PR DESCRIPTION
**Pull request by perry-mitchell**
This PR adds several features:
 * TOTP support
 * Field value types

image

image

### CODE REVIEW

File: src/components/vault/EntryDetails.js [hunk 1]
```diff
@@ -177,17 +208,41 @@ const FieldRow = ({ field, editing, onFieldNameUpdate, onFieldUpdate, onRemoveFi
     ) : (
       field.title || field.property
     );
+  const renderMenu = (
+    <Menu>
+      {/*
+        The following is in the parent level due to:
```
Comment (sallar) [inline src/components/vault/EntryDetails.js]: Damn

File: src/components/vault/EntryDetails.js [hunk 2]
```diff
@@ -240,7 +295,10 @@ const FieldRow = ({ field, editing, onFieldNameUpdate, onFieldUpdate, onRemoveFi
                 </Otherwise>
               </Choose>
               <If condition={field.removeable}>
-                <Button icon="trash" onClick={() => onRemoveField(field)} />
+                <Popover content={renderMenu} boundary="viewport" captureDismiss={false}>
+                  <Button icon="cog" />
+                </Popover>
+                {/* <Button icon="trash" onClick={() => onRemoveField(field)} /> */}
```
Comment (sallar) [inline src/components/vault/EntryDetails.js]: This can be deleted I guess?

File: src/components/vault/reducers/entry.js [hunk 3]
```diff
@@ -35,5 +35,19 @@ export function entryReducer(state, action) {
         })
       };
     }
+    case 'set-field-value-type': {
+      const { field: changedField, valueType } = action;
+      console.log('SET FIELD VALUE TYPE', changedField, valueType);
```
Comment (sallar) [inline src/components/vault/reducers/entry.js]: Also this?

File: src/components/vault/reducers/entry.js [hunk 4]
```diff
@@ -35,5 +35,19 @@ export function entryReducer(state, action) {
...
+      return {
+        ...state,
+        fields: state.fields.map(field => {
+          if (field.id === changedField.id) {
+            console.log('MATCHED', field);
```
Comment (sallar) [inline src/components/vault/reducers/entry.js]: And this

File: src/components/vault/EntryDetails.js (hunk 1 above)
Comment (perry-mitchell) [reply src/components/vault/EntryDetails.js]: ikr.. Oh well, we can check back with it later..

**Comment by perry-mitchell** [comment]
@sallar Changes made! Thanks

**Pull request merged**